### PR TITLE
[node] Update typings to v20.15.0

### PR DESCRIPTION
**Pull request by Semigradsky**
https://github.com/nodejs/node/releases/tag/v20.15.0

- test_runner: support test plans
- zlib: expose zlib.crc32()

### CODE REVIEW

**Comment by Semigradsky** [comment]
@jakebailey could you please take a look? There are just backported changes.

File: types/node/v20/package.json [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+    "version": "20.15.9999",
     "nonNpm": "conflict",
     "nonNpmDescription": "Node.js",
     "projects": [
         "https://nodejs.org/"
     ],
     "tsconfigs": ["tsconfig.dom.json", "tsconfig.non-dom.json"],
     "dependencies": {
-        "undici-types": "~5.26.4"
+        "undici-types": "~6.13.0"
```
Comment (jakebailey) [inline types/node/v20/package.json:12]: Did this actually get updated in 20.15, or is this correcting a mistake? I don't see this upgrade in the changelog (though 20.16 does get 6.19)

Comment (Semigradsky) [reply types/node/v20/package.json:12]: It was actually updated earlier, in [v20.13.0](https://github.com/nodejs/node/releases/tag/v20.13.0)

Comment (Semigradsky) [reply types/node/v20/package.json:12]: PR: https://github.com/nodejs/node/pull/52493

**Review by jakebailey — APPROVED** [review]

**Comment by Semigradsky** [comment]
Ready to merge

**Pull request merged**